platform/darwin/src/http_file_source.mm

```diff
@@ -196,24 +196,45 @@ void cancel() {
 
 HTTPFileSource::~HTTPFileSource() = default;
 
+MGL_EXPORT
+NSURL *resourceURLWithAccountType(const Resource& resource, NSInteger accountType) {
+    
+    NSURL *url = [NSURL URLWithString:@(resource.url.c_str())];
+    
+#if TARGET_OS_IPHONE || TARGET_OS_SIMULATOR
+    if (accountType == 0 &&
+        ([url.host isEqualToString:@"mapbox.com"] || [url.host hasSuffix:@".mapbox.com"])) {
```

Comment:
Is this basically checking whether the request URL is under the current or default base URL?

https://github.com/mapbox/mapbox-gl-native/blob/e79080771b36ad3a4be0dee8ca12032cd1c8488a/include/mbgl/util/constants.hpp#L62
https://github.com/mapbox/mapbox-gl-native/blob/1bef3b7d1d877c67db051b33e0516b36acee700d/include/mbgl/storage/default_file_source.hpp#L46

The only edge case I can think of is that a tileset may explicitly refer to tiles.mapbox.com via the http: or https: scheme instead of mapbox:.

Reply:
There's no change in this code - just pulled out for testing purposes. I'm not a huge fan of this check - hopefully in future we could do something different here.